mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1110,49 +1101,69 @@ def _log_custom_metric_artifact(self, artifact_name, raw_artifact, custom_metric
         artifact._load(artifact_file_local_path)
         return artifact
 
+    def _get_args_for_metrics(self, custom_metric, eval_df):
+        # deepcopying eval_df and builtin_metrics for each custom metric function call,
+        # in case the user modifies them inside their function(s).
+        eval_df_copy = eval_df.copy()
+        input_df = self.X.copy_to_avoid_mutation()
+        parameters = dict(inspect.signature(custom_metric.eval_fn).parameters)
+        eval_fn_args = {}
+        if "eval_df" in parameters:
```

Comment:
I wouldn't necessarily rely on the specific argument name `eval_df`. I think we can do this check:

If `eval_fn` defines two parameters, then we know it has the old format. If it defines 3+ parameters, it has the new format.